edgar-doctor/src/components/app/dashboardPages/appointments/AppointmentsCards.tsx

```diff
@@ -1,38 +1,63 @@
 'use client';
 
+import { useState } from 'react';
+
 import { Text, VStack } from '@chakra-ui/react';
 
 import AppointmentCard from 'components/app/dashboardPages/appointments/AppointmentCard';
 
+import Pagination from 'components/navigation/Pagination';
+import paginationHandler from 'utils/navigation/paginationHandler';
+import countMaxNumberPage from 'utils/navigation/countMaxNumberPage';
+
 import { type AppointmentType } from 'types/app/dashboard/appointments/appointmentType';
 
-const AppointmentsCards = ({ appointments }: { appointments: AppointmentType[] }): JSX.Element => (
-	<VStack spacing="32px" w="100%">
-		<VStack w="100%" spacing="16px" align="start">
-			<Text size="boldXl" id="edgar-dashboardAppointmentsPage-nextAppointments-text">
-				Mes prochains rendez-vous
-			</Text>
-			<VStack w="100%">
-				{appointments
-					.filter((appointment) => appointment.endDate > new Date().getTime())
-					.map((appointment) => (
+const AppointmentsCards = ({ appointments }: { appointments: AppointmentType[] }): JSX.Element => {
+	const [pageIndexNext, setPageIndexNext] = useState(1);
+	const [pageIndexOld, setPageIndexOld] = useState(1);
+	const upcomingAppointments = appointments.filter((appointment) => appointment.endDate > new Date().getTime());
+	const pastAppointments = appointments.filter((appointment) => appointment.endDate <= new Date().getTime());
+
+	return (
+		<VStack spacing="32px" w="100%">
+			<VStack w="100%" spacing="16px" align="start">
+				<Text size="boldXl" id="edgar-dashboardAppointmentsPage-nextAppointments-text">
+					Mes prochains rendez-vous
+				</Text>
+				<VStack w="100%">
+					{paginationHandler(upcomingAppointments, pageIndexNext, 3).map((appointment) => (
 						<AppointmentCard appointment={appointment} key={appointment.id} />
 					))}
+				</VStack>
+				{upcomingAppointments.length > 0 && (
+					<Pagination
+						pageIndex={pageIndexNext}
+						maxPageNumbers={countMaxNumberPage(appointments, 3)}
```

Comment:
It's not `appointments` but `upcomingAppointments` because your pagination handle the display of upcomingAppointments and not overall appointments